scripts/limits.js

```diff
@@ -0,0 +1,300 @@
+        });
+
+        await builder.moveCall({
+            target: `${packageId}::gateway::send_message`,
+            arguments: [
+                gateway,
+                message,
+            ]
+        });
+
+        await builder.signAndExecute(keypair);
+    }, 1000, 1000000);
+}
+
+async function getMaxApprovals() {
+    const message = {
+        source_chain: 'Ethereum',
+        source_address: 'Source Address',
+        destination_id: keccak256(defaultAbiCoder.encode(['string'], ['destination'])),
+        payload_hash: keccak256(defaultAbiCoder.encode(['string'], ['payload hash'])),
+    };
+
+    return await binarySearch(async (n) => {
+        const messages = Array.from({length: n}, (_, index) => {
+            return {
+                message_id: getRandomBytes32(),
+                ...message,
+            }
+        });
+        await approveMessage(client, keypair, gatewayInfo, messages);
+    });
+}
+
+async function prepare() {
+    client = new SuiClient({ url: getFullnodeUrl(network) });
+
+    await Promise.all(
+        [operator, deployer, keypair].map((keypair) =>
+            requestSuiFromFaucetV0({
+                host: getFaucetHost(network),
+                recipient: keypair.toSuiAddress(),
+            }),
+        ),
+    );
+
+    await publishPackage(client, deployer, 'utils');
+    await publishPackage(client, deployer, 'version_control');
+    let result = await publishPackage(client, deployer, 'axelar_gateway');
+    packageId = result.packageId;
+    const creatorCap = result.publishTxn.objectChanges.find(
+        (change) => change.objectType === `${packageId}::gateway::CreatorCap`,
+    ).objectId;
+    result = await publishPackage(client, deployer, 'relayer_discovery');
+    const discoveryPackageId = result.packageId;
+    discovery = result.publishTxn.objectChanges.find(
+        (change) => change.objectType === `${discoveryPackageId}::discovery::RelayerDiscovery`,
+    ).objectId;
+
+    gatewayInfo = {
+        ...calculateNextSigners(),
+        ... gatewayInfo,
+    }; 
+
+    const encodedSigners = WeightedSigners.serialize(gatewayInfo.signers).toBytes();
+    const builder = new TxBuilder(client);
+
+    await builder.moveCall({
+        target: `${packageId}::gateway::setup`,
+        arguments: [
+            creatorCap,
+            operator.toSuiAddress(),
+            domainSeparator,
+            minimumRotationDelay,
+            previousSignersRetention,
+            encodedSigners,
+            CLOCK_PACKAGE_ID,
+        ],
+    });
+
+    result = await builder.signAndExecute(deployer);
+
+    gateway = result.objectChanges.find((change) => change.objectType === `${packageId}::gateway::Gateway`).objectId;
+
+    gatewayInfo.gateway = gateway;
+    gatewayInfo.domainSeparator = domainSeparator;
+    gatewayInfo.packageId = packageId;
+    discoveryInfo.packageId = discoveryPackageId;
+    discoveryInfo.discovery = discovery;
+}
+
+(async() => {
+    await prepare();
```

Comment:
create a `main` function. avoid global variables at the top

Reply:
done but seems worse. Do you mean pass any variables to every function explicitly?